src/networking/clientMessageReceiver.java

```diff
@@ -0,0 +1,203 @@
+	 * @return Message
+	 */
+	public String getMsgFromPacket(String packet){
+
+		/**
+		 * Converts a String into json object to retreive the required values
+		 */
+		JSONObject json = new JSONObject(packet);
+
+		/**
+		 * Retreives the message from the json object
+		 */
+		String message = json.getString("message");
+
+		// Returns the message
+		return message;
+	}
+
+	/**
+	 * This method pushes the message into respective queues based on their identifier.
+	 * @param id Identifier of the message
+	 * @param msg Message that is to be transported over the network.
+	 */
+	public void push(String id, String msg) {
+
+		/**
+		 * Creates a object of queue type.
+		 */
+		IncomingPacket queuePacket = new IncomingPacket(msg, id);
+
+		/**
+		 * Checks whether to push the object into content module queue or processing module queue
+		 */
+		if(id.equals("content")){
+			contModuleQueue.enqueue(queuePacket);
+		}else{
+			procModuleQueue.enqueue(queuePacket);
+		}
+		logger.log(ModuleID.NETWORKING, LogLevel.SUCCESS, "Message pushed into appropriate queue by server based on the identifier");
+	}
+	
+	/**
+	 * 
+	 * This is the default method that would be executed when the thread is started
+	 * because the class implements Runnable interface.
+	 * This method receives the message from the server and pushes it into the respective Queue.
+	 * 
+	 */
+	@Override
+    public void run(){
+
+		/**
+		 * Loops until the socket gets closed by the internet communicator, when the socket gets closed
+		 * the appropriate catch would be executed.
+		 */
+		while(true){
+			/**
+			 * This block of code inside the try would try to execute instructions inside it and when it 
+			 * receives any exceptions it would look for appropriate catch block.
+			 */
+			try{
+
+				/**
+				 * Receives the input message from the input stream
+				 */
+				String recvMsg = dis.readUTF();
+				String id = getIdFromPacket(recvMsg);
+				String msg = getMsgFromPacket(recvMsg);
+
+				/**
+				 * Calls the push function
+				 */
+				push(id, msg);
+			}
+
+			/**
+			 * This block gets executed  if this stream reaches the end before reading all the bytes.
+			 */
+			catch(EOFException exp){
+				//Logs exception
+				logger.log(ModuleID.NETWORKING, LogLevel.WARNING, exp.toString());
+			}
+
+			/**
+			 * This block gets executed if the bytes do not represent a valid modified UTF-8 encoding of a string.
+			 */
+			catch(UTFDataFormatException exp){
+				//Logs exception
+				logger.log(ModuleID.NETWORKING, LogLevel.WARNING, exp.toString());
+			}
+
+			/**
+			 * This block gets executed the stream has been closed and the underlying input stream does not support 
+			 * reading after close, or another I/O error occurs.
+			 */
+			catch(IOException exp){
+				//Logs exception
+				logger.log(ModuleID.NETWORKING, LogLevel.WARNING, exp.toString());
+			}
+
+			/**
+			 * This block gets executed whether there is an exception or not in try block
+			 */		
+			finally{
```

Comment:
Remove this finally block as you are closing the stream after sending one message , add return statements in the corresponding exception blocks